src/expiryWarning.ts

```diff
@@ -0,0 +1,85 @@
+import moment from 'moment';
+import { promisify } from 'util';
+import { resolve, sep } from 'path';
+import fs from 'fs';
+const readdir = promisify(fs.readdir);
+const stat = promisify(fs.stat);
+
+import {
+  DEFAULT_RECORDING_DIR,
+  DEFAULT_EXPIRATION,
+  DEFAULT_DAYS_FOR_EXPIRY_WARNING,
+  COLORS,
+} from './constants';
+
+// TYPES
+interface recordingFile {
+  name: string,
+  status: string,
+}
+
+// CONSTANTS
+const DAYS_TO_WARN: number = process.env.DAYS_TO_WARN ? parseInt(process.env.DAYS_TO_WARN, 10) : DEFAULT_DAYS_FOR_EXPIRY_WARNING;
+const DAYS_EXPIRY: number = process.env.DAYS_EXPIRY ? parseInt(process.env.DAYS_EXPIRY, 10) : DEFAULT_EXPIRATION;
+const RECORDINGS_FOLDER: string = process.env.RECORDINGS_FOLDER || DEFAULT_RECORDING_DIR;
+const STATES = {
+  EXPIRED: `${COLORS.RED}Expired${COLORS.DEFAULT}`,
+  ALMOST_EXPIRED: `${COLORS.YELLOW}Expires in less than ${DEFAULT_DAYS_FOR_EXPIRY_WARNING} days${COLORS.DEFAULT}`,
+};
+
+// Get all expired or soon to be expired recordings
+export async function getFiles(dir: string, fileList: recordingFile[] = []) {
+  let subDirectories;
+  let currentFileList = fileList;
+  try {
+    subDirectories = await readdir(dir);
+  } catch (e) {
+    // eslint-disable-next-line no-console
+    console.error(`[Polly Jest Preset] Recordings folder "${dir}" not found`);
+    return [];
+  }
+  await Promise.all(subDirectories.map(async (directory: string) => {
+    const absolutePath = resolve(dir, directory);
+    if ((await stat(absolutePath)).isDirectory()) {
+      currentFileList = await getFiles(absolutePath, fileList)
```

Comment:
Furthermore, I'm a little skeptical that this return value will process correctly because you have all the branches of a `Promise.all` overwriting this single var. Don't see how any branch of the tree will roll up to the correct result.

Reply:
I still suspect there's a race condition here, though -- if the tree traversal is sufficiently deep (e.g. one promise takes 1s and another takes 5s) I think the last one will win, right?

Reply:
Just pushed an update